src/main/resources/db/migration/V5.3.0.3__7551-expanded-compound-datasetfield-validation.sql

```diff
@@ -0,0 +1,16 @@
+-- update required attribute of parents of compound fields where a subfield is required
+UPDATE datasetfieldtype SET required=true 
+WHERE id IN 
+(SELECT parentdatasetfieldtype_id FROM datasetfieldtype WHERE required=true);
+
+
+-- add input level rows for parents of compound fields  where a subfield has required input level
+-- (first run a delete to clear out any possible existing parent values)
+DELETE FROM dataversefieldtypeinputlevel WHERE datasetfieldtype_id in (
+SELECT parentdatasetfieldtype_id FROM datasetfieldtype dsft);
+  
+INSERT INTO dataversefieldtypeinputlevel (include,required,datasetfieldtype_id,dataverse_id)
+SELECT true, true, parentdatasetfieldtype_id, dataverse_id FROM
```

Comment:
Need a "DISTINCT" here (line 13) for cases where there are multiple datasetfieldtypes with the same parent that are set to required.
(I only caught this because I was playing around with input levels while working on another issue.)
Without the "DISTINCT" I couldn't deploy the branch to my local.

Reply:
Good catch; I think in all my testing I only had one child input level overridden.